src/wallet/coinselection.h

```diff
@@ -237,11 +243,37 @@ struct OutputGroup
         m_subtract_fee_outputs(params.m_subtract_fee_outputs)
     {}
 
-    void Insert(const COutput& output, size_t ancestors, size_t descendants, bool positive_only);
+    void Insert(const std::shared_ptr<COutput>& output, size_t ancestors, size_t descendants);
     bool EligibleForSpending(const CoinEligibilityFilter& eligibility_filter) const;
     CAmount GetSelectionAmount() const;
 };
 
+struct Groups {
+    // Stores 'OutputGroup' containing only positive UTXOs (value > 0).
+    std::vector<OutputGroup> positive_group;
+    // Stores 'OutputGroup' which may contain both positive and negative UTXOs.
+    std::vector<OutputGroup> mixed_group;
+};
+
+/** Stores several 'Groups' whose were mapped by output type. */
+struct OutputGroups
+{
+    // Map filters to output groups.
+    std::map<OutputType, Groups> groups_by_type;
+    // All groups, no filters
+    Groups all_groups;
```

Comment:
nit: `All` groups is somewhat misleading here, because these are actually filtered groups, but for all output types. Unfortunately, I don't have better suggestion yet.

Reply:
The "filter" term usage here in the comment is wrong. For what corresponds to this struct, the `all` term just refers to the combination of all inserted groups. Same as we do in the `CoinsResult::All` method. The filtering process, nor what is inserted (whether is filtered or not), is not responsibility of this struct.

It's just a container that keeps track of `Groups` by type and the caches the combination of all of them.

Reply:
I mean, it would be slower but could also be created on-demand:
```c++
Groups OutputGroups::All()
{
    Groups ret;
    for (const auto& [type, groups_vec] : groups_by_type) {
        ret.positive_group.insert(ret.positive_group.begin(), groups_vec.positive_group.begin(), groups_vec.positive_group.end());
        ret.mixed_group.insert(ret.mixed_group.begin(), groups_vec.mixed_group.begin(), groups_vec.mixed_group.end());
    }
    return ret;
}
```

Reply:
Fixed the "filter" comment mention.